### PR TITLE
Enable unified_mode

### PR DESCRIPTION
**Pull request by axl89**
# Description

Enabled `unified_mode` on all resources while preserving compatibility with lower versions of Chef.

## Issues Resolved

N/A

## Check List

- [x] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by axl89** [comment]
Any clue on the [CentOS tests failing](https://github.com/sous-chefs/vagrant/runs/3970061503?check_suite_focus=true)? I'm a bit clueless here :sweat_smile:

**Review by ramereth — DISMISSED** [review]
Can you also address the remaining cookstyle issues?

File: test/fixtures/cookbooks/test/recipes/generic.rb [hunk 1]
```diff
@@ -43,6 +43,10 @@
 package 'gvfs-fuse'
 package 'unzip'
 
+if platform_family?('rhel')
+  package 'glibc'
+end
+
```
[the remark this answers is not in the document]
Comment (axl89) [reply test/fixtures/cookbooks/test/recipes/generic.rb]: Reverted in 07bf33300be8433c86d9c55928c63b0435485148

**Comment by kitchen-porter** [comment]
Released as: [3.0.0](https://github.com/sous-chefs/vagrant/releases/tag/3.0.0)

**Pull request merged**